### PR TITLE
Throw error if it exists for dashboard variables

### LINKED ISSUE #101: Errors in template variable queries are not shown in the dashboard
When defining a template variable query with timestream, if an error exists nothing is shown in the dashboard and may behave inconsistently over time.

This may be general grafana issues, or template variable queries need to re-throw any exceptions.

### PR DESCRIPTION
**Pull request by andresmgot**
Fixes #101

Screenshot from 2021-11-26 13-21-06

In the future, we should replace this code and use a custom variable editor.

### CODE REVIEW

**Review by vickyyyyyyy — APPROVED** [review]
LGTM - would we be able to add a test for this?

File: src/DataSource.ts [hunk 1]
```diff
@@ -291,6 +291,9 @@ export class DataSource extends DataSourceWithBackend<TimestreamQuery, Timestrea
       range,
     } as unknown) as DataQueryRequest).pipe(
       map((res) => {
+        if (res.error) {
+          throw new Error(res.error.message);
```
Comment (vickyyyyyyy) [inline src/DataSource.ts:295]: would it be possible for `res.error` to not have a message? If so, should we handle that case with `res.error?.message`?

Comment (vickyyyyyyy) [reply src/DataSource.ts:295]: oops, I meant to check if there is a message, not that we want to throw an error with no message 😄

Comment (andresmgot) [reply src/DataSource.ts:295]: by definition yes, message is optional, but in practice don't think so

**Comment by andresmgot** [comment]
> LGTM - would we be able to add a test for this?

I wouldn't worry that much about it in this case. We should refactor this component and use a proper QueryEditor here so this code won't be used.

**Pull request merged**